### PR TITLE
Rewrite Amplifier plugin code

### PR DESCRIPTION
**Pull request by LostRobotMusic**
When I make new effects for LMMS, I (and I assume most others) generally copy+paste the Amplifier as a template.  It's extremely obnoxious for it to not follow current code style guidelines (and also be written in some really weird ways in the DSP specifically), so I rewrote it all to be much more concise and far closer to what it should be.

### CODE REVIEW

File: plugins/Amplifier/Amplifier.cpp [hunk 1]
```diff
@@ -50,99 +50,61 @@ Plugin::Descriptor PLUGIN_EXPORT amplifier_plugin_descriptor =
...
-
 extern "C"
 {
 
 // necessary for getting instance out of shared lib
-PLUGIN_EXPORT Plugin * lmms_plugin_main( Model* parent, void* data )
+PLUGIN_EXPORT Plugin * lmms_plugin_main(Model* parent, void* data)
 {
-	return new AmplifierEffect( parent, static_cast<const Plugin::Descriptor::SubPluginFeatures::Key *>( data ) );
+	return new AmplifierEffect(parent, static_cast<const Plugin::Descriptor::SubPluginFeatures::Key *>(data));
```
Comment (messmerd) [inline plugins/Amplifier/Amplifier.cpp:105]:
```suggestion
	return new AmplifierEffect(parent, static_cast<const Plugin::Descriptor::SubPluginFeatures::Key*>(data));
```
style

File: plugins/Amplifier/AmplifierControls.h [hunk 2]
```diff
@@ -39,34 +39,24 @@ namespace gui
...
 	Q_OBJECT
 public:
-	AmplifierControls( AmplifierEffect* effect );
+	AmplifierControls(AmplifierEffect* effect);
 	~AmplifierControls() override = default;
 
-	void saveSettings( QDomDocument & _doc, QDomElement & _parent ) override;
-	void loadSettings( const QDomElement & _this ) override;
+	void saveSettings(QDomDocument & doc, QDomElement & parent) override;
+	void loadSettings(const QDomElement & parent) override;
```
Comment (messmerd) [inline plugins/Amplifier/AmplifierControls.h:50]:
```suggestion
	void saveSettings(QDomDocument& doc, QDomElement& parent) override;
	void loadSettings(const QDomElement& parent) override;
```
style

**Review by messmerd — COMMENTED** [review]
Gave it a quick look-over

File: plugins/Amplifier/Amplifier.cpp [hunk 3]
```diff
@@ -50,99 +50,61 @@ Plugin::Descriptor PLUGIN_EXPORT amplifier_plugin_descriptor =
...
-		return( false );
-	}
+	if (!isEnabled() || !isRunning()) { return false ; }
 
 	double outSum = 0.0;
 	const float d = dryLevel();
 	const float w = wetLevel();
-
-	const ValueBuffer * volBuf = m_ampControls.m_volumeModel.valueBuffer();
+
```
Comment (DomClark) [inline plugins/Amplifier/Amplifier.cpp:67]: This line has some rogue whitespace in it.

File: plugins/Amplifier/Amplifier.cpp [hunk 4]
```diff
@@ -50,99 +50,61 @@ Plugin::Descriptor PLUGIN_EXPORT amplifier_plugin_descriptor =
...
-	}
+	if (!isEnabled() || !isRunning()) { return false ; }
 
 	double outSum = 0.0;
 	const float d = dryLevel();
 	const float w = wetLevel();
-
-	const ValueBuffer * volBuf = m_ampControls.m_volumeModel.valueBuffer();
+	
+	const ValueBuffer * volumeBuf = m_ampControls.m_volumeModel.valueBuffer();
```
Comment (messmerd) [inline plugins/Amplifier/Amplifier.cpp:68]: The placement of the asterisk for these four variables can be fixed:
```cpp
const ValueBuffer* foo
```

File: plugins/Amplifier/Amplifier.cpp [hunk 5]
```diff
@@ -50,99 +50,61 @@ Plugin::Descriptor PLUGIN_EXPORT amplifier_plugin_descriptor =
...
-		const float left2 = leftBuf
-			? leftBuf->value( f )
-			: m_ampControls.m_leftModel.value();
-		const float right2 = rightBuf
-			? rightBuf->value( f )
-			: m_ampControls.m_rightModel.value();
-
-		s[0] *= left1 * left2 * 0.01;
-		s[1] *= right1 * right2 * 0.01;
+		s[0] *= volume * left * panLeft;
```
Comment (messmerd) [inline plugins/Amplifier/Amplifier.cpp:85]: I believe the array `s` could be removed and the whole calculation could be simplified to just:
```cpp
buf[f][0] *= d + w * volume * left * panLeft;
buf[f][1] *= d + w * volume * right * panRight;
```

Comment (DomClark) [reply plugins/Amplifier/Amplifier.cpp:85]: While that would be simpler, I think it's actually clearer how it currently is. I like that the wet value is calculated first, then the buffer set to a mixture of wet and dry. With this change, the factors for the wet value are split across both sides of the assignment, with the mixing code in between, and the wet value itself never materialises - you have to expand the product mentally to see how the code works.

**Review by sakertooth — APPROVED** [review]
LGTM

**Comment by messmerd** [comment]
The `#endif // LMMS_AMPLIFER_H` header guard closing comments could be added to.

**Review by DomClark — APPROVED** [review]

**Review by messmerd — APPROVED** [review]

**Pull request merged**